userchroot.c

```diff
@@ -631,9 +684,17 @@ int main(int argc, char* argv[], char* envp[]) {
 
     if (strncmp("--install-devices",argv[2],17) == 0) {
       rc = create_fundamental_devices(final_path);
+      rc |= create_mount_directory(final_path, "/dev/shm", (0777 | S_ISVTX));
```

Comment:
I would use the more restrictive `S_IRWXU` here as well. After mounting, the mode of the underlying mount point is irrelevant anyway, so better not allow arbitrary users to write into the not yet mounted directory.